### PR TITLE
Fix/positionduplication

### PR DESCRIPTION
**Pull request by kiliangrashoff**
Fixes the duplication of position between our model, the pieceViewMatrix in the GUIController, and a position variable within PieceView.

### CODE REVIEW

File: src/main/java/nl/tudelft/ti2206/semgr8/model/Piece.java [hunk 1]
```diff
@@ -46,6 +50,14 @@ public static Piece getRandomPiece(int x, int y) {
         return new Piece(x,y,(new Random()).nextInt(AMOUNT_OF_COLORS));
```
Comment (albert-smit) [inline src/main/java/nl/tudelft/ti2206/semgr8/model/Piece.java]: please run an autoformatter

File: src/main/java/nl/tudelft/ti2206/semgr8/view/GUIController.java [hunk 2]
```diff
@@ -50,22 +44,10 @@ public void init(Board board) {
     }
 
     public void swap(SwapData swapData) {
```
Comment (albert-smit) [inline src/main/java/nl/tudelft/ti2206/semgr8/view/GUIController.java]: please add javadoc

File: src/main/java/nl/tudelft/ti2206/semgr8/view/GUIController.java [hunk 3]
```diff
@@ -50,22 +44,10 @@ public void init(Board board) {
...
-        pieceViewMatrix[x1][y1] = pieceViewMatrix[x2][y2];
-        pieceViewMatrix[x2][y2] = temp;
-
-        boardGrid.setConstraints(pieceViewMatrix[x1][y1], x1, y1);
-        pieceViewMatrix[x1][y1].setPosition(p1);
-        boardGrid.setConstraints(pieceViewMatrix[x2][y2], x2, y2);
-        pieceViewMatrix[x2][y2].setPosition(p2);
+        Piece p1 = swapData.getPiece1();
+        Piece p2 = swapData.getPiece2();
+        boardGrid.setConstraints(p1.getView(), p1.getPosition().getX(), p1.getPosition().getY());
```
Comment (albert-smit) [inline src/main/java/nl/tudelft/ti2206/semgr8/view/GUIController.java]: this is a static method

File: src/main/java/nl/tudelft/ti2206/semgr8/view/PieceView.java [hunk 4]
```diff
@@ -25,13 +24,9 @@
      *
      * @param p
      *            the piece to be shown
-     * @param pos
-     *            the position of the piece
      */
-    public PieceView(final SuperPiece p, final Position pos, final InputEventListener e) {
+    public PieceView(final Piece p, final InputEventListener e) {
```
Comment (albert-smit) [inline src/main/java/nl/tudelft/ti2206/semgr8/view/PieceView.java]: please update this javadoc

File: src/main/java/nl/tudelft/ti2206/semgr8/view/GUIController.java [hunk 5]
```diff
@@ -34,15 +29,14 @@
      */
     public void init(Board board) {
         int gridDimension = board.getGridDimensions();
-        pieceViewMatrix = new PieceView[gridDimension][gridDimension];
         for (int i = 0; i < gridDimension; i++) {
             for (int j = 0; j < gridDimension; j++) {
-                SuperPiece piece = board.getPiece(i, j);
+                Piece piece = board.getPieceExternal(i, j);
                 Position pos = new Position(i, j);
```
Comment (albert-smit) [inline src/main/java/nl/tudelft/ti2206/semgr8/view/GUIController.java]: this variable is no longer used

**Comment by albert-smit** [comment]
looks good when tests and comments are fixed

**Comment by kiliangrashoff** [comment]
changes processed, thanks for the feedback!

**Pull request merged**